libs/etterlatte-ktor/src/main/kotlin/HttpCallLogging.kt

```diff
@@ -0,0 +1,69 @@
+package no.nav.etterlatte.libs.ktor
+
+import io.ktor.client.HttpClient
+import io.ktor.client.plugins.HttpClientPlugin
+import io.ktor.client.request.HttpSendPipeline
+import io.ktor.client.statement.HttpReceivePipeline
+import io.ktor.util.AttributeKey
+import net.logstash.logback.marker.Markers
+import org.slf4j.LoggerFactory
+import java.util.concurrent.TimeUnit
+
+internal class HttpCallLogging private constructor() {
+    private val logger = LoggerFactory.getLogger(HttpCallLogging::class.java)
+    private val requestStartTime = AttributeKey<Long>("requestStartTime")
+
+    // Kun dummy pga krav
+    class Config
+
+    companion object : HttpClientPlugin<Config, HttpCallLogging> {
+        override val key: AttributeKey<HttpCallLogging> = AttributeKey("GjennyHttpCallLogging")
+
+        override fun prepare(block: Config.() -> Unit): HttpCallLogging {
+            return HttpCallLogging()
+        }
+
+        override fun install(
+            plugin: HttpCallLogging,
+            scope: HttpClient,
+        ) {
+            plugin.setupPreRequestHandler(scope)
+            plugin.setupPostRequestLogHandler(scope)
+        }
+    }
+
+    private fun setupPreRequestHandler(client: HttpClient) {
+        client.sendPipeline.intercept(HttpSendPipeline.Before) {
+            context.attributes.put(requestStartTime, System.nanoTime())
+            return@intercept
+        }
+    }
+
+    private fun setupPostRequestLogHandler(client: HttpClient) {
+        client.receivePipeline.intercept(HttpReceivePipeline.State) { res ->
+            with(res.call) {
+                val requestStartTime = attributes[requestStartTime]
+                val executionTime = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - requestStartTime)
+
+                logger.info(
+                    Markers.appendEntries(
+                        mapOf<String, Any?>(
+                            "x_upstream_host" to request.url.host,
+                            "x_request_uri" to request.url.encodedPath,
+                            "x_request_method" to request.method,
```

Comment:
Jeg ville testet dette litt i dev for å se at output blir korrekt. Mener å huske jeg fikk litt rar output ved å prefikse med `x_` da jeg satt opp logging i frontend siden logstash legger på `x_` automatisk i enkelte tilfeller. Mulig den bare gjør det dersom det er en nøkkel som er "reservert", men skal ikke si det for sikkert 🤔

Reply:
Det er også min erfaring når jeg la på `correlation_id`. Tror alle brukerdefinerte variabler får ett sånt prefix.

Reply:
Will do, har søkt på typ "request" og "user" i kibana og ser det kommer opp mange rare kombinasjoner.